File: 백준/Silver/10845. 큐/큐.py
# 큐 # 선입선출

import sys
n = int(input())
queue = []

for i in range(n):
    user = sys.stdin.readline().split()

    if user[0] == 'push':
        queue.append(user[1])

    if user[0] == 'pop': # 큐에서 가장 앞에 있는 정수를 빼
        if len(queue) == 0:
            print(-1)
        else:
            print(queue.pop(0))
    
    if user[0] == 'size':
        print(len(queue))
    
    if user[0] == 'empty':
        if len(queue) == 0:
            print(1)
        else:
            print(0)
    
    if user[0] == 'front': # 큐의 가장 앞에 있는 정수
        if len(queue) == 0:
            print(-1)
        else:
            print(queue[0])

    if user[0] == 'back': # 큐의 가장 뒤에 있는 정수
        if len(queue) == 0:
            print(-1)
        else:
            print(queue[-1])

File: 백준/Silver/10845. 큐/큐.js
let [ N,...input ] = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n").map(v => v.split(" "));

const queue = [];
let ans = '';
for(let i = 0; i < N; i++) {
    //
    switch(input[i][0]){
        case "push":
            queue.push(input[i][1]);
            break;
        case "pop":
            ans += (queue.length === 0 ? "-1" : queue.shift()) + "\n";
            break;
        case "size":
            ans += (queue.length) + "\n";
            break;
        case "empty":
            ans += (queue.length === 0 ? "1" : "0") + "\n";
            break;
        case "front":
            ans += (queue.length === 0 ? "-1" : queue[0]) + "\n"; 
            break;
        case "back":
            ans += (queue.length === 0 ? "-1" : queue[queue.length - 1]) + "\n";
            break;
    };    
};
console.log(ans);